### PR TITLE
Add passwordless-related information to protos

### PR DESCRIPTION
**Pull request by codingllama**
Introduces webauthn.User (for usernameless logins) and extends webauthn.SessionData and types.WebauthnDevice with relevant fields.

#9160

### CODE REVIEW

**Comment by codingllama** [comment]
It's a short one (diff is mainly generated files). I'll use the new info in upcoming PRs.

File: api/types/webauthn/webauthn.proto [hunk 1]
```diff
@@ -49,6 +49,19 @@ message SessionData {
     bytes user_id = 2 [ (gogoproto.jsontag) = "userId,omitempty" ];
     // Raw Credential IDs of the credentials allowed for the ceremony.
     repeated bytes allow_credentials = 3 [ (gogoproto.jsontag) = "allowCredentials,omitempty" ];
+    // True if resident keys were required by the server / Relying Party.
+    bool resident_key = 4;
+    // Requested user verification requirement, either "discouraged" or
+    // "required".
+    // An empty value is treated equivalently to "discouraged".
+    string user_verification = 5;
```
Comment (smallinsky) [inline api/types/webauthn/webauthn.proto:57]: Thus values are know maybe enum type will fit better here ?

File: api/types/webauthn/webauthn.proto [hunk 2]
```diff
@@ -49,6 +49,19 @@ message SessionData {
     bytes user_id = 2 [ (gogoproto.jsontag) = "userId,omitempty" ];
     // Raw Credential IDs of the credentials allowed for the ceremony.
     repeated bytes allow_credentials = 3 [ (gogoproto.jsontag) = "allowCredentials,omitempty" ];
+    // True if resident keys were required by the server / Relying Party.
+    bool resident_key = 4;
```
Comment (smallinsky) [inline api/types/webauthn/webauthn.proto:53]: missing filed tags.

Comment (codingllama) [reply api/types/webauthn/webauthn.proto:53]: Well spotted, done.

File: api/types/webauthn/webauthn.proto (hunk 1 above)
Comment (codingllama) [reply api/types/webauthn/webauthn.proto:57]: I like the more "opaque" value here, since we don't own the underlying protocol (WebAuthn).

**Comment by codingllama** [comment]
PTAL?

File: api/types/webauthn/webauthn.proto (hunk 1 above)
Comment (smallinsky) [reply api/types/webauthn/webauthn.proto]: All right, I'm ok with current version. We call always use `casttype` in the future if needed.

**Review by smallinsky — APPROVED** [review]

**Review by ibeckermayer — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks folks, merging asap.

**Pull request merged**